File: frontend/src/types/document.ts
export type DocumentStatus = 'uploaded' | 'processing' | 'ready' | 'failed'
export type EmbeddingStatus = 'not_started' | 'pending' | 'processing' | 'embedded' | 'failed'

export interface DocumentItem {
  id: string
  filename: string
  status: DocumentStatus
  size_bytes: number
  created_at: string
  page_count: number
  chunk_count: number
  extracted_char_count: number
  error_message: string | null
  embedding_status: EmbeddingStatus
  embedded_chunk_count: number
  embedding_model: string | null
  embedding_error: string | null
  embedded_at: string | null
}

export interface DocumentChunk {
  id: string
  document_id: string
  chunk_index: number
  page_number: number
  text: string
  char_count: number
  created_at: string
  embedding_status: EmbeddingStatus
  embedding_model: string | null
  embedding_error: string | null
  embedded_at: string | null
}

export interface DocumentChunksResponse {
  document_id: string
  chunks: DocumentChunk[]
}

File: backend/src/app/schemas/document.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

DocumentStatus = Literal["uploaded", "processing", "ready", "failed"]
EmbeddingStatus = Literal["not_started", "pending", "processing", "embedded", "failed"]


class DocumentResponse(BaseModel):
    id: str
    filename: str
    status: DocumentStatus
    size_bytes: int
    created_at: datetime
    page_count: int
    chunk_count: int
    extracted_char_count: int
    error_message: str | None
    embedding_status: EmbeddingStatus
    embedded_chunk_count: int
    embedding_model: str | None
    embedding_error: str | None
    embedded_at: datetime | None


class DocumentChunkResponse(BaseModel):
    id: str
    document_id: str
    chunk_index: int
    page_number: int
    text: str
    char_count: int
    created_at: datetime
    embedding_status: EmbeddingStatus
    embedding_model: str | None
    embedding_error: str | None
    embedded_at: datetime | None


class DocumentChunksResponse(BaseModel):
    document_id: str
    chunks: list[DocumentChunkResponse]
